lib/Alien/Build/Plugin/Build/Autoconf.pm

```diff
@@ -234,6 +238,67 @@ Some reasonable default flags will be provided.
   $self;
 }
 
+sub _pkgconf_wrapper {
+  my ($self, $build) = @_;
+
+  return if !_win;
+
+  my $pk = File::Which::which ($ENV{PKG_CONFIG})
+        || File::Which::which ('ppkg-config')
+        || File::Which::which ('pkg-config.pl')
+        || File::Which::which ('pkg-config');
+
+  if (!defined $pk) {
+    $build->log ("Could not locate ppkg-config or pkg-config in your path:\n");
+    return;
+  }
+  
+  
+  if ($pk =~ /\.bat$/i) {
+    $pk =~ s/\.bat$//i;
+    #  should not die?
+    die "bat file does not have adjacent pkg-config"
+      if !-e $pk;
+  }
+  
+  #  flaky means of determining we are using a perl pkg-config
+  my $data = Path::Tiny->new($pk)->slurp;
+  my $str = <<"EOSTRING"
+# lightweight no-dependency version of pkg-config. This will work on any machine
+# with Perl installed.
+EOSTRING
+  ;
+
+  my $fname;  
+  if ($data =~ /$str/msi) {
+    my $perl = $^X;
+    $perl =~ s/\.exe$//i;
+    foreach my $path ($perl, $pk) {
+      $path =~ s{\\}{/}g;
+      $path =~ s{^([a-z]):/}{/$1/}i;
+    }
+    my $args = '$' . join ' $', (1..9);
+    
+    my $wrapper = <<"EOWRAPPER"
+#/bin/sh
+
+$perl $pk $args
+EOWRAPPER
+  ;
+    $build->log ("Pure perl pkg-config detected on windows.\n");
+    $build->log ("Wrapping $pk in shell script to cope with MSYS perl and paths.\n");
+    $fname = Path::Tiny->new($build->root, 'pkg-config');
```

Comment:
Put this in a temp directory under the build root, like here:
https://github.com/Perl5-Alien/Alien-Build/pull/195/files#diff-9ce385ad3e86640b003d212edf3a4d7bL158

Reply:
Will do.

Reply:
Done in c6f71bf4175c7b8085894bd3c1736952705042b1